lib/collection/src/shards/replica_set.rs

```diff
@@ -871,7 +871,15 @@ impl ShardReplicaSet {
         let remotes = self.remotes.read().await;
 
         let (local, is_local_ready) = match self.local.try_read() {
-            Ok(local) => (futures::future::ready(local).left_future(), true),
+            Ok(local) => {
+                let is_local_ready = local
+                    .deref()
+                    .as_ref()
+                    .map_or(false, Shard::is_update_in_progress);
```

Comment:
This probably should be inverted: 
```suggestion
                    .map_or(false, |shard| !shard.is_update_in_progress());
```

Reply:
Thanks! Sorry. 🤦‍♀️